File: learn the basics/maths/primenum.py
import math
def primecheck(n):
    if(n==1):
        return False
    if(n==2 or n==3):
        return True
    if(n%2==0 or n%3==0):
        return False
    for i in range(5,int(math.sqrt(n))+1,6):
        if n%i==0 or n%(i+2)==0:
            return False
    return True
n = int(input("Enter a number: "))

if primecheck(n):
    print(f"{n} is a prime number.")
else:
    print(f"{n} is not a prime number.")

File: learn the basics/maths/primenum.cpp
#include <iostream>
#include <cmath>
using namespace std;

// Function to check if a number is prime
bool primecheck(int n) {
    if (n == 1) {
        return false;
    }
    if (n == 2 || n == 3) {
        return true;
    }
    if (n % 2 == 0 || n % 3 == 0) {
        return false;
    }
    for (int i = 5; i <= sqrt(n); i += 6) {
        if (n % i == 0 || n % (i + 2) == 0) {
            return false;
        }
    }
    return true;
}

int main() {
    int n;
    cout << "Enter a number: ";
    cin >> n;

    if (primecheck(n)) {
        cout << n << " is a prime number." << endl;
    } else {
        cout << n << " is not a prime number." << endl;
    }

    return 0;
}
